components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TiersApiServiceImpl.java

```diff
@@ -58,66 +62,113 @@
+    public Response tiersTierLevelGet(String tierLevel, Integer limit, Integer offset, String accept,
+            String ifNoneMatch) {
 
         //pre-processing
         //setting default limit and offset if they are null
         limit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
         offset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
         try {
-            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
-            Set<Tier> tiers = apiProvider.getTiers();
             List<Tier> tierList = new ArrayList<>();
-            if (tiers != null)
-                tierList.addAll(tiers);
-            TierListDTO tierListDTO = TierMappingUtil.fromTierListToDTO(tierList, limit, offset);
-            TierMappingUtil.setPaginationParams(tierListDTO, limit, offset, tierList.size());
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+            if (!StringUtils.isBlank(tierLevel)) {
+                if (TierDTO.TierLevelEnum.api.toString().equals(tierLevel)) {
+                    Set<Tier> apiTiers = apiProvider.getTiers(APIConstants.TIER_API_TYPE, tenantDomain);
+                    if (apiTiers != null) {
+                        tierList.addAll(apiTiers);
+                    }
+                } else if (TierDTO.TierLevelEnum.application.toString().equals(tierLevel)){
+                    Set<Tier> appTiers = apiProvider.getTiers(APIConstants.TIER_APPLICATION_TYPE, tenantDomain);
+                    if (appTiers != null) {
+                        tierList.addAll(appTiers);
+                    }
+                } else if (TierDTO.TierLevelEnum.resource.toString().equals(tierLevel)){
+                    Set<Tier> resourceTiers = apiProvider.getTiers(APIConstants.TIER_RESOURCE_TYPE, tenantDomain);
+                    if (resourceTiers != null) {
+                        tierList.addAll(resourceTiers);
+                    }
+                } else {
+                    throw RestApiUtil.buildNotFoundException(
+                            "tierLevel should be one of " + Arrays.toString(TierDTO.TierLevelEnum.values()));
+                }
+            } else {
+                throw RestApiUtil.buildBadRequestException("tierLevel cannot be empty");
+            }
+            TierListDTO tierListDTO = TierMappingUtil.fromTierListToDTO(tierList, tierLevel, limit, offset);
+            TierMappingUtil.setPaginationParams(tierListDTO, tierLevel , limit, offset, tierList.size());
             return Response.ok().entity(tierListDTO).build();
         } catch (APIManagementException e) {
             String errorMessage = "Error while retrieving tiers";
-            log.error(errorMessage, e);
-            throw new InternalServerErrorException(e);
+            handleException(errorMessage, e);
         }
+        return null;
```

Comment:
Unreachable return statement. Looks like the error is repeated in multiple places. Please fix all if valid.